helix-core/src/main/java/org/apache/helix/ConfigAccessor.java

```diff
@@ -1009,4 +1010,19 @@ public ConfigAccessor build() {
               _zkAddress), false);
     }
   }
+
+  /**
+   * Validate if the topology related settings (Domain or ZoneId) in the given instanceConfig
+   * are valid and align with current clusterConfig.
+   * This function should be called when instance added to cluster or caller updates instanceConfig.
+   *
+   * @throws IllegalArgumentException
+   */
+  public static boolean validateTopologySettingInInstanceConfig(ClusterConfig clusterConfig,
```

Comment:
Seems this method is not used by ConfigAccessor itself?  My suggestion is put this method into InstanceConfig (or ClusterConfig) and it does not have to be static (it can be called like myInstanceConfig.validateTopologySetting(String clusterTopologyString).

Reply:
Updated.